homeassistant/components/switch/wirelesstag.py

```diff
@@ -0,0 +1,129 @@
+from homeassistant.const import (
+    CONF_ENTITY_NAMESPACE, CONF_MONITORED_CONDITIONS, STATE_OFF, STATE_ON)
+import homeassistant.helpers.config_validation as cv
+
+DEPENDENCIES = ['wirelesstag']
+
+_LOGGER = logging.getLogger(__name__)
+
+ARM_TEMPERATURE = 'temperature'
+ARM_HUMIDITY = 'humidity'
+ARM_MOTION = 'motion'
+ARM_LIGHT = 'light'
+ARM_MOISTURE = 'moisture'
+
+# Switch types: Name, tag sensor type
+SWITCH_TYPES = {
+    ARM_TEMPERATURE: ['Arm Temperature', 'temperature'],
+    ARM_HUMIDITY: ['Arm Humidity', 'humidity'],
+    ARM_MOTION: ['Arm Motion', 'motion'],
+    ARM_LIGHT: ['Arm Light', 'light'],
+    ARM_MOISTURE: ['Arm Moisture', 'moisture']
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_ENTITY_NAMESPACE, default=DEFAULT_ENTITY_NAMESPACE):
+        cv.string,
+    vol.Required(CONF_MONITORED_CONDITIONS, default=[]):
+        vol.All(cv.ensure_list, [vol.In(SWITCH_TYPES)]),
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up switches for a Wireless Sensor Tags."""
+    platform = hass.data.get(WIRELESSTAG_DOMAIN)
+
+    switches = []
+    tags = platform.load_tags()
+    for switch_type in config.get(CONF_MONITORED_CONDITIONS):
+        for _, tag in tags.items():
+            if switch_type in WirelessTagSwitch.allowed_switches(tag):
+                switches.append(WirelessTagSwitch(platform, tag, switch_type))
+
+    add_devices(switches, True)
+
+
+class WirelessTagSwitch(WirelessTagBaseSensor, SwitchDevice):
+    """A switch implementation for Wireless Sensor Tags."""
+
+    @classmethod
+    def allowed_switches(cls, tag):
+        """Return allowed switch types for wireless tag."""
+        all_sensors = SWITCH_TYPES.keys()
+        sensors_per_tag_spec = {
+            WIRELESSTAG_TYPE_13BIT: [
+                ARM_TEMPERATURE, ARM_HUMIDITY, ARM_MOTION],
+            WIRELESSTAG_TYPE_WATER: [
+                ARM_TEMPERATURE, ARM_MOISTURE],
+            WIRELESSTAG_TYPE_ALSPRO: [
+                ARM_TEMPERATURE, ARM_HUMIDITY, ARM_MOTION, ARM_LIGHT]
+        }
+
+        tag_type = tag.tag_type
+
+        result = (
+            sensors_per_tag_spec[tag_type]
+            if tag_type in sensors_per_tag_spec else all_sensors)
+        _LOGGER.info("Allowed switches: %s tag_type: %s",
+                     str(result), tag_type)
+
+        return result
+
+    def __init__(self, api, tag, switch_type):
+        """Initialize a switch for Wireless Sensor Tag."""
+        super().__init__(api, tag)
+        self._switch_type = switch_type
+        self.sensor_type = SWITCH_TYPES[self._switch_type][1]
+        self.define_name('{} {}'.format(self._tag.name,
+                                        SWITCH_TYPES[self._switch_type][0]))
+        self._api.register_entity(self)
+
+    @property
+    def assumed_state(self):
+        """Return true if unable to access real state of entity."""
+        return False
+
+    @property
+    def state(self):
```

Comment:
Do not override `state`, just implement `is_on` property

Reply:
@baloob I have state defined in my base class, so if I do not override state here, switch does not work as expected.

Reply:
got it. reworking...